exampleShared/java/net/gini/android/vision/example/RuntimePermissionHandler.java

```diff
@@ -0,0 +1,276 @@
+package net.gini.android.vision.example;
+
+import android.Manifest;
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.provider.Settings;
+
+import com.karumi.dexter.Dexter;
+import com.karumi.dexter.PermissionToken;
+import com.karumi.dexter.listener.DexterError;
+import com.karumi.dexter.listener.PermissionDeniedResponse;
+import com.karumi.dexter.listener.PermissionGrantedResponse;
+import com.karumi.dexter.listener.PermissionRequest;
+import com.karumi.dexter.listener.PermissionRequestErrorListener;
+import com.karumi.dexter.listener.single.PermissionListener;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Created by Alpar Szotyori on 05.12.2017.
+ *
+ * Copyright (c) 2017 Gini GmbH.
+ */
+
+public class RuntimePermissionHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RuntimePermissionHandler.class);
+
+    private final Activity mActivity;
```

Comment:
I'm doubting here. Could it be a problem if this property is `strong`?

Reply:
I really appreciate that you point out these possible leaks! 👍 

In this case it's not an issue because the permission handler is owned by the activity only. The activity reference is passed only to Dexter and AlertDialogs. I've seen in Dexter's source, that they use weak references.